pkg/image/validation/kubernetes_test.go

```diff
@@ -1079,3 +1070,691 @@ func TestValidateAdditionalArtifacts(t *testing.T) {
+	}
+
+	b, err := yaml.Marshal(serverConfig)
+	require.NoError(t, err)
+
+	configFile := filepath.Join(serverConfigDir, "server.yaml")
+	require.NoError(t, os.WriteFile(configFile, []byte(b), 0o600))
+
+	failures := validateNetworkingConfig(&k8s, configFile)
+
+	assert.Len(t, failures, 0)
+}
+
+func mockValidateConfig(k8s *image.Kubernetes, serverConfig map[string]any) []FailedValidation {
+	var failures []FailedValidation
+
+	failures = append(failures, validateNodeIP(k8s, serverConfig)...)
+	failures = append(failures, validateCIDRConfig(k8s, serverConfig)...)
+
+	return failures
+}
+
+func TestValidateConfig(t *testing.T) {
+	tests := map[string]struct {
+		K8s                    image.Kubernetes
+		ServerConfig           map[string]any
+		ExpectedFailedMessages []string
+	}{
+		`valid IPv6 prio`: {
+			K8s: image.Kubernetes{
+				Network: image.Network{
+					APIVIP4: "192.168.1.1",
+					APIVIP6: "fd12:3456:789a::21",
+				},
+			},
+			ServerConfig: map[string]any{
+				"cluster-cidr": "fd12:3456:789b::/48,10.42.0.0/16",
+				"service-cidr": "fd12:3456:789c::/112,10.43.0.0/16",
+			},
+		},
+		`invalid both IPv4`: {
+			K8s: image.Kubernetes{
+				Network: image.Network{
+					APIVIP4: "192.168.1.1",
+					APIVIP6: "fd12:3456:789a::21",
+				},
+			},
+			ServerConfig: map[string]any{
+				"cluster-cidr": "10.42.0.0/16,10.44.0.0/16",
+				"service-cidr": "10.43.0.0/16,10.45.0.0/16",
+			},
+			ExpectedFailedMessages: []string{
+				"Kubernetes server config cluster-cidr cannot contain addresses of the same IP address family; one must be IPv4, and the other IPv6",
+				"Kubernetes server config service-cidr cannot contain addresses of the same IP address family; one must be IPv4, and the other IPv6",
+			},
+		},
+		`invalid both IPv6`: {
+			K8s: image.Kubernetes{
+				Network: image.Network{
+					APIVIP4: "192.168.1.1",
+					APIVIP6: "fd12:3456:789a::21",
+				},
+			},
+			ServerConfig: map[string]any{
+				"cluster-cidr": "fd12:3456:789d::/48,fd12:3456:789b::/48",
+				"service-cidr": "fd12:3456:789e::/112,fd12:3456:789c::/112",
+			},
+			ExpectedFailedMessages: []string{
+				"Kubernetes server config cluster-cidr cannot contain addresses of the same IP address family; one must be IPv4, and the other IPv6",
+				"Kubernetes server config service-cidr cannot contain addresses of the same IP address family; one must be IPv4, and the other IPv6",
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			k := test.K8s
+			sc := test.ServerConfig
+			failures := mockValidateConfig(&k, sc)
+			assert.Len(t, failures, len(test.ExpectedFailedMessages))
+
+			var foundMessages []string
+			for _, foundValidation := range failures {
+				foundMessages = append(foundMessages, foundValidation.UserMessage)
+			}
+
+			for _, expectedMessage := range test.ExpectedFailedMessages {
+				assert.Contains(t, foundMessages, expectedMessage)
+			}
+
+		})
+	}
+}
+
+func TestValidateCIDRConfig(t *testing.T) {
```

Comment:
We're missing the following cases:
- No CIDRs when API VIPs are unset
- More than two addresses in a given CIDR field